### PR TITLE
dev-lang/rust: add patch to support LibreSSL 2.8.x and 2.9.0

### PR DESCRIPTION
**Pull request by steils**
Patches from the upstream:
https://github.com/sfackler/rust-openssl/commit/9fd7584a84168655cb27e03b7e19a9847b88e77f
https://github.com/sfackler/rust-openssl/commit/af4488357c9b3e003b883e89c16aaa675ad0c6ac

Closes: https://bugs.gentoo.org/677302
Package-Manager: Portage-2.3.60, Repoman-2.3.12
Signed-off-by: Stefan Strogin <stefan.strogin@gmail.com>

### CODE REVIEW

File: dev-lang/rust/files/1.32.0-libressl.patch [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
...20008ad90dc346a7dcc02f9201d94bff0bb7d1d71031c097850287c50","src/pkcs7.rs":"5370c8d4bfbf1502fa33aff635cb730e7c87814580f1df9bdb1412a492377813","src/rand.rs":"156e605ecb45d4e1b7a7c404003ccd19c4a2226cbfe47938ad1dce240ed4369f","src/rsa.rs":"a0bf64f1f8ff9f6e5dbed6c38d8fa2d3d1c5dfcfe6dd3ff79f7737b072ce4681","src/safestack.rs":"6c39e28565d34efad707d77561d4caa99e3f028fcac3a2ef6fd403a78de1190c","src/sha.rs":"05044d221d8d205010a4e3760c0b5c0e2e923e165db0d65f9a8c9bcedaad9961","src/srtp.rs":"306047f3f31828fe7d15de154b375f071ae07012a94c0b5c909f5fe57962a11b","src/ssl.rs":"e69d3b9b8f2f8ae1ed3905a4dbe064ad1b7c1eaf21d7eabb86ecdb5b41e2120a","src/ssl3.rs":"9336c816e00847d552dea22587d4ac72ff3cbd469fa5ff750423a19ea11e68eb","src/stack.rs":"1a509907283e5a2bf88cf193ce607f49ce7d2d95547c2ce2abc0fd4567aad334","src/tls1.rs":"fd85a1683064def0a9e07d803b90102229adc1d58fa19cf4e56976f0dd7425b9","src/x509.rs":"839be2feb74755ab7df66a024ad27dfa70ce4f7889229d9152ba999f3a54b914","src/x509_vfy.rs":"5609f6300ab37a4d4755c81f7af1e6bb93a2f1a6f333c6e42f1e73ebf83ddc05","src/x509v3.rs":"66fd646ea4caf5ce81e3c881e88d85462519c4e97a7f82e82ca8ac52ea830ad2"},"package":"ff3d1b390ab1b9700f682ad95a30dc9c0f40dd212ca57266012cfc678b0e365a"}
```
Comment (cardoe) [inline dev-lang/rust/files/1.32.0-libressl.patch]: This appears to be just a new line at the end change?

File: dev-lang/rust/files/1.32.0-libressl.patch [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
++++ b/vendor/openssl-sys/build/main.rs
+@@ -1,9 +1,9 @@
+ extern crate cc;
++#[cfg(feature = "vendored")]
++extern crate openssl_src;
+ extern crate pkg_config;
+ #[cfg(target_env = "msvc")]
+ extern crate vcpkg;
+-#[cfg(feature = "vendored")]
+-extern crate openssl_src;
```
Comment (cardoe) [inline dev-lang/rust/files/1.32.0-libressl.patch]: Why does this ordering matter?

File: dev-lang/rust/files/1.32.0-libressl.patch [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
++        )
++    }
+ }
+ 
+ #[cfg(not(feature = "vendored"))]
+ mod imp {
+     use pkg_config;
+-    use std::path::{Path, PathBuf};
+     use std::ffi::OsString;
++    use std::path::{Path, PathBuf};
```
Comment (cardoe) [inline dev-lang/rust/files/1.32.0-libressl.patch]: Why the re-ordering here?

File: dev-lang/rust/files/1.32.0-libressl.patch [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+-	}
++    use openssl_src;
++    use std::path::PathBuf;
++
++    pub fn get_openssl(_target: &str) -> (PathBuf, PathBuf) {
++        let artifacts = openssl_src::Build::new().build();
++        (
++            artifacts.lib_dir().to_path_buf(),
++            artifacts.include_dir().to_path_buf(),
++        )
```
Comment (cardoe) [inline dev-lang/rust/files/1.32.0-libressl.patch]: Functionally this is identical?

File: dev-lang/rust/files/1.32.0-libressl.patch [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
++    use std::path::{Path, PathBuf};
+     use std::process::{self, Command};
+ 
+     use super::env;
+@@ -500,6 +503,8 @@ See rust-openssl README for more information:
+             (7, _) => ('7', 'x'),
+             (8, 0) => ('8', '0'),
+             (8, 1) => ('8', '1'),
++            (8, _) => ('8', 'x'),
++            (9, 0) => ('9', '0'),
```
Comment (cardoe) [inline dev-lang/rust/files/1.32.0-libressl.patch]: This is the only functional change. Let's drop all the other movement and just keep this.

File: dev-lang/rust/files/1.32.0-libressl.patch (hunk 1 above)
Comment (steils) [reply dev-lang/rust/files/1.32.0-libressl.patch]: No, this like changes sha256sum for `main.rs` from `f509c63668156ce48b93d05d5483a5bfb112dc8f786ff6296f7801add2db7452` to `c8bc352b94d05afeaeb96300341a695c70c3b62c46793a5c035aaf42e22b69d1`.
No other changes to `.cargo-checksum.json`, all other checksums are left the same.

File: dev-lang/rust/files/1.32.0-libressl.patch (hunk 2 above)
Comment (steils) [reply dev-lang/rust/files/1.32.0-libressl.patch]: That's from https://github.com/sfackler/rust-openssl/commit/9fd7584a84168655cb27e03b7e19a9847b88e77f
The patch is called "Support LibreSSL 2.8.x", I don't know if these lines order matters.

File: dev-lang/rust/files/1.32.0-libressl.patch (hunk 3 above)
Comment (steils) [reply dev-lang/rust/files/1.32.0-libressl.patch]: The same here. Just applied https://github.com/sfackler/rust-openssl/commit/9fd7584a84168655cb27e03b7e19a9847b88e77f

**Comment by steils** [comment]
Done. Removed the code movement.

**Pull request closed without merging**